tests/testthat/test-try_stack.R

```diff
@@ -25,6 +25,16 @@ describe("with_retries", {
       "not enough retries"
     )
   })
-
-
 })
+
+describe("try_with_exit_code", {
+    test_that("it evaluates expr successfully", {
+    expect_true(try_with_exit_code(sum(1, 2)))
+  })
+    test_that("Expr evaluation fails and it exits R with specific exit code", {
+    with_mock(
+      `q` = function(status, ...) stop(status),
+      expect_error(try_with_exit_code(sum("someval"), exit_code = 300))
```

Comment:
I know `expect_error` has a grep method for the error msg, but I can't seem to match it.
Trying 
```   
 with_mock(
      `q` = function(status, ...) stop(status),
      expect_error(try_with_exit_code(sum("someval"), exit_code = 300), "300")
    )
```

Gives `Error: invalid 'type' (character) of argument\n"`

Yet, if I do `expect_error(stop(300), "300")`, character argument works. Any suggestions?